File: Java Script - Introduction to Programming/5.1. Loops/10. Half Sum Element.js
function halfSumElement(arg) {
  arg = arg.map(Number)
  arg.shift()
  let sortedList = arg.sort((a, b) => a - b)
  let biggestNumber = sortedList.pop()
  let allSum = sortedList.reduce((a, b) => a + b)
  return allSum === biggestNumber ? `Yes\nSum = ${biggestNumber}` : `No\nDiff = ${Math.abs(allSum - biggestNumber)}`
}

//console.log(halfSumElement([7, 3, 4, 1, 1, 2, 12, 1]))
// console.log(halfSumElement([4, 6, 1, 2, 3]))

File: Pyrhon - Introduction to Programming/5.1. Loops/10. Half Sum Element.py
sorted_num = sorted([int(input()) for _ in range(int(input()))])
big_num = sorted_num.pop()
sum_all_num = sum(sorted_num)
print(f'Yes\nSum = {big_num}' if sum_all_num == big_num else f'No\nDiff = {abs(sum_all_num - big_num)}')


'''
3
5
5
1
'''
